### PR TITLE
core: Refactor Java code

### PR DESCRIPTION
**Pull request by Moriadry**
Remove unnecessary 'continue' statement and simplify conditional expression

### CODE REVIEW

**Comment by dapengzhang0** [comment]
ok to test

**Comment by Moriadry** [comment]
improve performance issues.

File: core/src/main/java/io/grpc/ServerProvider.java [hunk 1]
```diff
@@ -39,7 +39,6 @@ static final ServerProvider load(ClassLoader cl) {
 
     for (ServerProvider current : providers) {
       if (!current.isAvailable()) {
-        continue;
```
Comment (ejona86) [inline core/src/main/java/io/grpc/ServerProvider.java]: The continue is more clear to me, and it is weird to have the block with no contents. It's honestly fine either way, so I think we should just leave it the way that it is.

File: core/src/main/java/io/grpc/internal/AbstractServerStream.java [hunk 2]
```diff
@@ -116,7 +116,7 @@ public final void writeHeaders(Metadata headers) {
   public final void deliverFrame(WritableBuffer frame, boolean endOfStream, boolean flush) {
     // Since endOfStream is triggered by the sending of trailers, avoid flush here and just flush
     // after the trailers.
-    abstractServerStreamSink().writeFrame(frame, endOfStream ? false : flush);
+    abstractServerStreamSink().writeFrame(frame, !endOfStream && flush);
```
Comment (ejona86) [inline core/src/main/java/io/grpc/internal/AbstractServerStream.java]: I find the previous version more clear that the `flush` is pass-through. I'm having trouble with this new version, although I do question why it's harder to comprehend.

File: core/src/main/java/io/grpc/ServerProvider.java (hunk 1 above)
Comment (Moriadry) [inline core/src/main/java/io/grpc/ServerProvider.java]: ok, your explanation makes sense, I will rever it.

**Pull request merged**